Profiler/CProfilerCallback.cpp

```diff
@@ -359,41 +347,44 @@ int CProfilerCallback::writeFileVersionInfo(LPCWSTR assemblyPath, char* buffer,
 
 HRESULT CProfilerCallback::JITCompilationFinished(FunctionID functionId,
 	HRESULT hrStatus, BOOL fIsSafeToBlock) {
-	FunctionInfo info;
-	getFunctionInfo(functionId, &info);
+	recordFunctionInfo(&jittedMethods, LOG_KEY_JITTED, functionId);
+	return S_OK;
+}
 
-	if (isEagerMode) {
-		writeSingleFunctionInfoToLog(LOG_KEY_JITTED, info);
-	}
-	else {
-		// Notify monitor that method has been jitted.
-		jittedMethods.push_back(info);
+HRESULT CProfilerCallback::JITInlining(FunctionID callerID, FunctionID calleeId,
+		BOOL* pfShouldInline) {
+	// Save information about inlined method (if not already seen)
+	if (inlinedMethodIds.insert(calleeId).second == true) {
+		recordFunctionInfo(&inlinedMethods, LOG_KEY_INLINED, calleeId);
 	}
 
-	// Always return OK
+	// Always allow inlining.
+	*pfShouldInline = true;
+
 	return S_OK;
 }
 
-HRESULT CProfilerCallback::JITInlining(FunctionID callerID, FunctionID calleeId,
-		BOOL* pfShouldInline) {
+void CProfilerCallback::recordFunctionInfo(std::vector<FunctionInfo>* recordedFunctionInfos, const char* logKey, FunctionID calleeId) {
 	FunctionInfo info;
 	getFunctionInfo(calleeId, &info);
 
-	if (isEagerMode) {
-		writeSingleFunctionInfoToLog(LOG_KEY_INLINED, info);
-	}
-	else {
-		// Save information about inlined method.
-		if (inlinedMethodIds.insert(calleeId).second == true) {
-			inlinedMethods.push_back(info);
+	EnterCriticalSection(&criticalSection);
+
+	if (eagerness == 1) {
+		// Directly write to log if we want to record each function immediatelly

```

Comment:
is this still an acutal use case that we have for any customer? otherwise i'd treat eagerness == 1 the same as any other eagerness > 0

Reply:
as mentioned, this is solely for performance reasons. removed this now. if you think we should keep, please let me know.